scripts/linters/pylint_extensions.py

```diff
@@ -1893,6 +1893,204 @@ def visit_compare(self, node):
                 self.add_message('inequality-with-none', node=node)
 
 
+class DisallowedFunctionCallsChecker(checkers.BaseChecker):
+    """Custom pylint checker for language specific general purpose
+    regex checks of functions calls to be removed or replaced.
+    """
+
+    __implements__ = interfaces.IAstroidChecker
+    name = 'disallowed-function-calls'
+    priority = -1
+    msgs = {
+        'C0031': (
+            'Please do not use %s.',
+            'remove-disallowed-function-calls',
+            'Please remove disallowed function calls.',
+        ),
+        'C0032': (
+            'Please replace %s with %s.',
+            'replace-disallowed-function-calls',
+            'Please replace disallowed function' +
+            'calls with allowed alternative.',
+        ),
+    }
+    python_utils_file = b'python_utils.py'
+    python_utils_test_file = b'python_utils_test.py'
+
+    # pylint: disable=disallowed-function-calls,single-line-pragma
+    options = ((
+        'disallowed-functions',
+        {
+            'default': (
+                (
+                    re.compile(r'datetime.datetime.now\(\)'),
+                    b'datetime.datetime.utcnow()',
+                    (),
+                    ()),
+                (
+                    re.compile(r'self.assertEquals\('),
+                    b'self.assertEqual()',
+                    (),
+                    ()),
+                (
+                    re.compile(r'StringIO\('),
+                    b'python_utils.string_io()',
+                    (python_utils_file, python_utils_test_file,),
```

Comment:
Instead of excluding the entire file should we use the pragama? (Ditto elsewhere)

Reply:
This is definitely a better idea, but keep in mind the pragmas would have to be made per-line since we won't be able to specify which specific function call is disallowed.

Reply:
Yeah, I think that's fine! Do you see any issue with that?

Reply:
Not that I'm aware of